lib/utilization/aws-info.js

```diff
@@ -24,27 +25,45 @@ function fetchAWSInfo(agent, callback) {
     return setImmediate(callback, null, results)
   }
 
-  const instanceHost = '169.254.169.254'
-  const apiVersion = '2016-09-02'
-  const endpoint = 'dynamic/instance-identity/document'
-  const url = 'http://' + instanceHost + '/' + apiVersion + '/' + endpoint
-  common.request(url, agent, function getMetadata(err, data) {
+  const authTokenOpts = {
+    headers: { 'X-aws-ec2-metadata-token-ttl-seconds': '21600' },
+    host: INSTANCE_HOST,
+    method: 'PUT',
+    path: '/latest/api/token',
+    timeout: 500
```

Comment:
I might just update the code in common.js to use 500ms instead of 1second for `setTimeout` so we aren't setting multiples. Or replace the setTimeout with just adding a timeout handler and leave these but both feels funky.

Reply:
k i cleaned this up and added generic unit tests to test the timeout.  FWIW the tests before weren't accurate